### PR TITLE
Four new comics

### PR DESCRIPTION
**Pull request by ExTechOp**
This adds four new comics, which I've now tested for a few days:
* backtobc (John Hart's **B.C.** back from when it was superb)
* kevinandkell (Bill Holbrook's **Kevin & Kell**, the world's longest-running daily webcomic)
* sequentialart (Phillip "Jolly Jack" M. Jackson's **Sequential Art**)
* viiviandwagner (Jussi "Juba" Tuomola's **Viivi and Wagner**, English-language version of _viivijawagner_)

Closes [`tardypad` issue \#121](https://github.com/tardypad/sailfishos-daily-comics/issues/121) (request for Sequential Art).

### CODE REVIEW

**Comment by Olf0** [comment]
Thank you very much, @ExTechOp!

In order to keep things organised (IOW: to prevent me from becoming confused):
* Please let us finish PR #108, first!
  There are a few open questions which arose from my review last week.  Please answer them briefly, I don't mind if you or me then implement corresponding changes.
* In general (but do *not* care about this here: it will be fine in the git-history, if I merge #108 first):
  If you want to contribute something unrelated (here: adding 4 new strips), please start anew from the default branch (i.e. `devel` for Daily Comics): You can either synchronise the `devel` branch in your cloned repository with "upstream" (i.e. the one of `sailfishos-applications`) and start from that (this method is the most easy one when working with a local clone of a remote repository), or start in GitHub's web-frontend editing something in the default ranch of "upstream".
  This way the commits from PR #108 would not have shown up in PR #111; but as stated before, all is fine when I merge #108 first.

**Comment by ExTechOp** [comment]
> In order to keep things organised (IOW, to prevent me from becoming confused):
> 
>     * Please let us finish PR [Add 5 new strips, fix one old one #108](https://github.com/sailfishos-applications/daily-comics/pull/108), first!
>       There are a few open questions which arose from my review last week.  Please answer them briefly, I don't mind if you or me then implement corresponding changes.

I'm sorry, I really don't quite see what questions still are open there, can you rephrase them?

>     * In general (but do _not_ care about this here: It will be fine in the git-history, if I merge [PR #108](https://github.com/sailfishos-applications/daily-comics/pull/108) first, but we must not add any additional commits for [PR #108](https://github.com/sailfishos-applications/daily-comics/pull/108) to this branch):
>       If you want to contribute something unrelated to a former PR and its branch (here: adding 4 new strips), please start anew from the default branch (i.e. `devel` for Daily Comics): You can either synchronise the `devel` branch in your cloned repository with "upstream" (i.e. the one of `sailfishos-applications`) and start from that (this method is the most easy one when working with a local clone of a remote repository), or start in GitHub's web-frontend editing something in the default branch of "upstream".
>       This way the commits from [PR #108](https://github.com/sailfishos-applications/daily-comics/pull/108) would not have shown up in PR [Four new comics #111](https://github.com/sailfishos-applications/daily-comics/pull/111); but as stated before, all is fine when I merge [PR #108](https://github.com/sailfishos-applications/daily-comics/pull/108) first.

Sorry, I will keep that in mind in the future. As I said, I haven't had time to really get into the nitty-gritty bits of how github is supposed to be used.

**Comment by Olf0** [comment]
Last thing first:

> I haven't had time to really get into the nitty-gritty bits of how github is supposed to be used.

Never mind about *"really get[ting] into the nitty-gritty bits of […] [git and] github['s web-frontend]"*, learning by doing is sufficient I can tell after a few years.

Hence I am trying to provide some advice which helps you to get along well quicker (i.e. advice which I would have been glad to receive half a decade ago).

In this context:
> I'm sorry, I really don't quite see what questions still are open there

I utilised GitHub's "review" functionality for these questions, which is a little awkward to use, but makes discussing multiple locations of a PR in parallel much easier:
It allows to quote a small section from a PR (at least four lines are displayed, usually only the last line of these is relevant, as it is the case with my four review-comments for PR #108) and have a conversation about exactly this part of a PR (i.e. a discussion sub-thread).

I wrote four review-comments, but only the first two contain real questions, the first one a single question, the second one two (feel free to also add comments to the third an fourth review-comment, if you like):
1. https://github.com/sailfishos-applications/daily-comics/pull/108#review-thread-or-comment-id-1174043601
2. https://github.com/sailfishos-applications/daily-comics/pull/108#review-thread-or-comment-id-1174044019

The links I provided here are just jump-marks (i.e. jumping to HTML `id` tags) within the `Conversation` "tab" of PR #108, so that the page is displayed scrolled down exactly as much to make the place where the `id` tag is embedded at the top of the browser window.  Side note: Sometimes browsers scroll a little to far and one has to scroll very little up to see tha start of the section which was meant to jump to.

If you follow each of these links (e.g. open them in new browser-tabs), on top of the browser window you should see four lines quoted from the PR, below that my comment WRT the last of these four lines and then an empty text field (at first just a single line field, which expands as you type), where you should input your reply.

If you still face issues using that as GitHub designed it, then you might take a look at [this description with screenshots](https://github.com/sailfishos-applications/flowplayer/pull/106#issuecomment-2497938868) (another software, PR and contributor, the same usability issue).

**Comment by Olf0** [comment]
Something seems to go wrong with GitHub's "review" functionality lately.  I order to proceed, I just made screenshots of my four review comments and [uploaded them on the "Conversation"-page of PR \#108](https://github.com/sailfishos-applications/daily-comics/pull/108#issuecomment-2510369525).

P.S.: Maybe some JavaScript filter (I use NoScript and uBlockOrigin) is in the way? JavaScript must be allowed for github.com (plus fetch and ping) and githubassets.com (plus fetch).

**Comment by Olf0** [comment]
All right, now that I merged PR #108 into `devel` and then updated this PR #111 with the recent commits (from PR #108) from `devel`, this branch `four_new_comics` only differs from `devel` by the … four new comics.  Hence I can now review these changes without having the changes of PR #ä108 intermingled.  Will do, when I have some time.

**Comment by Olf0** [comment]
My review:
1. Please HTTP**S**-ify all links which do not explicitly require HTTP as transport protocol (which you did so nicely in the past):
   * `backtobc`
   * `viiviandwagner`
2. `kevinandkell`: Why is the returned string statically prefixed with the `stripSource`?  I.e. is simply returning the `match[1]` not sufficient here?

I altered all `color` values, because…
* some were all white (`ffffff`).
* some where the same.

**Comment by Olf0** [comment]
What about using this as `cover.jpg` for "Back to B.C."?
| Original JPEG ([link](https://assets.gocomics.com/uploads/features/cbc/widgets/link_wo_title_large_cbc_bc_201706281341.jpg)) | Cropped PNG | JPEG scaled to 300x300 px² |
|---|---|---|
|  |  | link_wo_title_large_cbc_bc_201706281341_cropped_scaled |

**Comment by Olf0** [comment]
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Developing&nbsp;a&nbsp;new&nbsp;cover&nbsp;image&nbsp;for&nbsp;"Kevin&nbsp;&&nbsp;Kell" | &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; |
|---|---|
| From https://en.wikifur.com/w/images/6/63/KevinandKell.png or http://images.wikia.com/furry/images/6/63/KevinandKell.png<br />KevinandKell-logo | Background set transparent:<br />KevinandKell-logo_bg-transparent<br/>**Worse that route 2, below: Thus not used!** |
| From https://www.kevinandkell.com/images/headerLogo-2024b.png **and** https://www.kevinandkell.com/images/navbarContainerBackground-2024b.png<br />kevinandkell_headerLogo-2024b<br />**and**<br />kevinandkell_navbarContainerBackground-2024b | Stitched together:<br />kevinandkell_headerLogo-2024b_stiched<br />Continued next line … |
| Logo removed:<br />kevinandkell_Figures-2024b | K&K figures removed:<br />&nbsp;<br />kevinandkell_Logo-2024b |
| Logo and figures rearranged on square canvas:<br />(**preview with low-quality logo**)<br />KevinandKell-cover_bg-transparent | Background gradient from https://www.kevinandkell.com/images/navbarBackground.png:<br /><br />kevinandkell_navbarBackground |
| Background set to darkest green of gradient:<br />(**preview with low-quality logo**)<br />KevinandKell-cover_bg-green | Background set to green gradient:<br />(**preview with low-quality logo**)<br />KevinandKell-cover_bg-green-gradient |
| JPEG scaled to 300x300 px²:<br />(**preview with low-quality logo**)<br />KevinandKell-cover_bg-green | JPEG scaled to 300x300 px²:<br />(**preview with low-quality logo**)<br />KevinandKell-cover_bg-green-gradient |

**Comment by Olf0** [comment]
Oh, I had the impression something was missing and re-read my "[Adding a new comic](https://github.com/sailfishos-applications/daily-comics/blob/devel/info/comic_addition.md#adding-a-new-comic)" document:
- Please [create a `cover_big.jpg` file](https://github.com/sailfishos-applications/daily-comics/blob/devel/info/comic_addition.md#additional-requirements) for each comic in this PR from your original graphic material for the corresponding `cover.jpg` file; just upscaling the 300x300 px² `cover.jpg` to 1000x1000 px² does not make much sense and creates ugly compression artefacts be re-compression.
  For keeping that "original graphic material" (i.e. not format-converted or scaled, but freed of author names, copyright notices, superfluous graphic elements etc.) for cover and example images, I once created [`.xdata/plugins`](https://github.com/sailfishos-applications/daily-comics/tree/devel/.xdata/plugins), because of [issue \#65](https://github.com/sailfishos-applications/daily-comics/issues/65).  Hence please commit such "original graphic material" it there.
  
  I have already generated such 1000x1000 px² `cover_big.jpg` images for ["Back to B.C."](https://github.com/sailfishos-applications/daily-comics/pull/111#issuecomment-2525265433) and ["Kevin & Kell"](https://github.com/sailfishos-applications/daily-comics/pull/111#issuecomment-2525323154) above, but have not committed them to your [ExTechOp:four_new_comics](https://github.com/ExTechOp/daily-comics/tree/four_new_comics) branch.
  
  Please also create a separate pull-request branching from [the current `devel` branch](https://github.com/sailfishos-applications/daily-comics/tree/devel/) as a follow-up to PR #108 with the `cover_big.jpg` files for the four comics dealt with there, and extant "original graphic material".
- As you seem to have a local `git` clone of this repository on a Linux machine, please regenerate the comic covers grid with the `resources/generate_covers_grid.sh` shell script and 
regenerate the comics list with the `resources/generate_list.sh` shell script.  Preferably in a separate PR after this one and all the `cover_big.jpg` files have been merged.
  Thanks!

**Comment by Olf0** [comment]
Dear @ExTechOp, I would really like to create a new release with all your (and a few of my prior) changes, but some tasks remain to be carried out before that: Please see https://github.com/sailfishos-applications/daily-comics/pull/111#issuecomment-2526385940 and then the two prior messages of mine.

Thanks!

**Comment by ExTechOp** [comment]
My apologies for my tardiness, I've had a very hectic time at work.
Please do commit your **backtobc** and **kevinandkell** images, they are much nicer than what I had come up with.
As I've said I'm not a very sophisticated github user. I now created new _cover.jpg_ and _cover_big.jpg_ files for **bizarro** and **sequentialcomics** and tried to push my changes, do they look up they ended up correctly on this branch? I'll do the rest if :ok:

**Comment by ExTechOp** [comment]
Also, where are you coming up with these rather high-res images, I'm having a hard time finding anything suitable for **viiviandwagner** ?

**Comment by Olf0** [comment]
> My apologies for my tardiness, I've had a very hectic time at work.

As this is all voluntary spare time activity, that is absolutely O.K.  I have enough tasks at hand, that I was not actively waiting for a response and action, but simply wanted to avoid forgetting about this open thread (this applies to both of us, you and me).

> Please do commit your **backtobc** and **kevinandkell** images, they are much nicer than what I had come up with.

O.K., will do.

> As I've said I'm not a very sophisticated github user. I now created new _cover.jpg_ and _cover_big.jpg_ files for **bizarro** and **sequentialcomics** and tried to push my changes, do they look up they ended up correctly on this branch?

Yes, you are and were doing fine.

> I'll do the rest if 🆗

:ok:

**Comment by Olf0** [comment]
> > Please do commit your **backtobc** and **kevinandkell** images, they are much nicer than what I had come up with.
> 
> O.K., will do.

Done.

Of the essential data, only `plugins/viiviandwagner/cover_big.jpg` is still missing.

Additionally I would appreciate very much if you to add your source-pictures and -links plus critical intermediate steps in creating the cover image in [`.xdata/plugins`](https://github.com/ExTechOp/daily-comics/tree/four_new_comics/.xdata/plugins) for Sequential Art, Viivi & Wagner, and Bizarro, like I did for Back to B.C. and Kevin & Kell.  Thanks!

**Comment by Olf0** [comment]
Damn, the prior two commit message should have been "`Use exctraction RegEx as discussed in PR `[`#125`](https://github.com/sailfishos-applications/daily-comics/pull/125)", at least the detailed message points to the correct PR comment.

**Comment by ExTechOp** [comment]
Could either of these actually form a better basis for making a cover for Viivi and Wagner?
* https://gocomicscmsassets.gocomics.com/staging-assets/assets/Global_Feature_Badge_Viivi_and_Wagner_600_d2ce0ce56c.png
* https://www.riemumuseot.fi/uploads/2021/08/269c557d-juba-viivi-ja-wagner.jpg

I say go ahead and publish GoComics without Helsingin Sanomat, as their current version seems to run on obfuscated javascript, which will take quite some time to understand.

**Comment by Olf0** [comment]
> Could either of these actually form a better basis for making a cover for Viivi and Wagner?
> * https://gocomicscmsassets.gocomics.com/staging-assets/assets/Global_Feature_Badge_Viivi_and_Wagner_600_d2ce0ce56c.png 
> * https://www.riemumuseot.fi/uploads/2021/08/269c557d-juba-viivi-ja-wagner.jpg

While the second one is suitable as a basis for a cover page, the result will be worse in terms of artistic design and representing the comic well.  Hence I would rather stay with the general design of the current one, but re-create it out of the original material and export a `cover.jpeg` and a `cover_big.jpeg` from it.

Similarly for "Sequential Art": I am not satisfied with the cover image, as it does have more than a single border colour, but would leave it as is for the next release.  In order to rectify that later, please provide pointers to the original material you used.

Bluntly and directly asked: I do not comprehend, why you cannot retrace what you did when you originally created these cover images ("Sequential Art" and "Viivi & Wagner")?  I assume it started with an internet search (using search engine X and search terms Y) which provided links to the original material you used for creating these cover images.  What poses such a hurdle to simply do that again, and you will likely find the same images, again?  Provide me with the image URLs and I will do the rest (i.e. finish this PR #111) some day in order to proceed with this. 

> I say go ahead and publish GoComics without Helsingin Sanomat, as their current version seems to run on obfuscated javascript, which will take quite some time to understand.

No, the three non-functional Helsingin Sanomat comics from PR #108 will stay as is (i.e. I will not invest additional work to omit them): There are so many broken comics and I will explicitly mark the next release as a beta version, again.  Consequently all files of these plug-ins will be there for you to research a new extraction script for them.

It is rather this PR #111 I want to conclude before the next release, and the four comics from PR #108 which share the same issues: No `cover_big.jpg` and no (links) to the original graphic material usable / used for creating `cover.jpg` / `cover_big.jpg`.
